### PR TITLE
eloquent models 10/31 resolves #22 when complete

### LINKED ISSUE #22: Eloquent models for our tables
We need Eloquent models for the following tables (see https://laravel.com/docs/9.x/eloquent-relationships#defining-relationships):
- Game
- Category
  - many-to-many with Game
- Publisher
  - one-to-many with Game
- Wishlist (our many-to-many game_user table)
  - see https://laravel.com/docs/9.x/eloquent-relationships#defining-custom-intermediate-table-models
- note: category_game does not really warrant its own model

### PR DESCRIPTION
**Pull request by 12bitsample**
Hey Andrew! Okay so I created the models that issue #22 requested, and defined the relationships. However, I didn't see anywhere that I needed to add any values for the __construct() methods unless they were previously defined. 

Please take a look when you get the chance and let me know if any changes are deemed necessary, otherwise let me know if it is okay to assign someone to get this PR approved and merged! Thanks!

### CODE REVIEW

File: app/Models/Publisher.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Publisher extends Model
+{
+    /**
+     * Get the Publisher
+     */
+    public function __construct()
```
Comment (oldskoolfan) [inline app/Models/Publisher.php:12]: constructor needs removed

File: app/Models/Game.php [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 namespace App\Models;
 
-class Game
+use Illuminate\Database\Eloquent\Model;
+class Game extends Model
 {
     public function __construct(
```
Comment (oldskoolfan) [inline app/Models/Game.php:8]: constructor needs removed

File: app/Models/Category.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Category extends Model
+{
+    /**
+     * Get the Category
+     */
+    public function __construct()
```
Comment (oldskoolfan) [inline app/Models/Category.php:12]: constructor needs removed

File: app/Models/Publisher.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    {
+
+    }
+
+    public function game() {
```
Comment (oldskoolfan) [inline app/Models/Publisher.php:17]: this should be `games` because a publisher can have many games

File: app/Models/Wishlist.php [hunk 5]
```diff
@@ -16,4 +16,9 @@ public function toString(): string
     {
         return "Welcome to your wishlist! Hopefully you get $this->title";
     }
+
```
Comment (oldskoolfan) [inline app/Models/Wishlist.php:19]: this class is our Pivot table between users and games, so it needs to extend the Pivot class as mentioned here: https://laravel.com/docs/9.x/eloquent-relationships#defining-custom-intermediate-table-models

Also we need to add a games property to the Users table that uses this Pivot class, as mentioned in the above link.

File: app/Models/Wishlist.php [hunk 6]
```diff
@@ -16,4 +16,9 @@ public function toString(): string
...
+    public function game_user_table()
```
Comment (oldskoolfan) [inline app/Models/Wishlist.php:20]: we don't need this

**Review by oldskoolfan — CHANGES_REQUESTED** [review]
Yes there are some changes that need to be made. The two main things to note are:
1. Follow the Pivot class documentation here: https://laravel.com/docs/9.x/eloquent-relationships#defining-custom-intermediate-table-models
2. Eloquent models shouldn't have constructors, otherwise they won't behave as expected.

File: app/Models/Game.php (hunk 2 above)
Comment (12bitsample) [reply app/Models/Game.php:8]: resolved!

File: app/Models/Publisher.php (hunk 4 above)
Comment (12bitsample) [reply app/Models/Publisher.php:17]: resolved!

File: app/Models/Category.php (hunk 3 above)
Comment (12bitsample) [reply app/Models/Category.php:12]: resolved!

File: app/Models/Wishlist.php (hunk 5 above)
Comment (12bitsample) [reply app/Models/Wishlist.php:19]: Okay preeeeetty sure this is resolved!

File: app/Models/Wishlist.php (hunk 6 above)
Comment (12bitsample) [reply app/Models/Wishlist.php]: resolved!

File: app/Models/Wishlist.php [hunk 7]
```diff
@@ -17,7 +17,7 @@ public function toString(): string
         return "Welcome to your wishlist! Hopefully you get $this->title";
     }
 
-    public function game_user_table()
+    public function pivot()
```
Comment (oldskoolfan) [inline app/Models/Wishlist.php:20]: i htink you can remove this function altogether

File: app/Models/User.php [hunk 8]
```diff
@@ -41,4 +41,8 @@ class User extends Authenticatable
     protected $casts = [
         'email_verified_at' => 'datetime',
     ];
+
+    public function games() {
+        return $this->belongsToMany(Game::class)->using(Wishlist::class);
+    }
```
Comment (mdigrandi) [inline app/Models/User.php:47]: Great job on this btw!

**Review by oldskoolfan — APPROVED** [review]
let's try to remove that pivot function on the Wishlist class at some point

**Pull request merged**